infra-gen2/backends/storage/dots-in-name/amplify/backend.ts

```diff
@@ -0,0 +1,39 @@
+import { defineBackend } from "@aws-amplify/backend";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import { v4 as uuidv4 } from "uuid";
+import { auth } from "./auth/resource";
+import { storage } from "./storage/resource";
+
+/**
+ * @see https://docs.amplify.aws/react/build-a-backend/ to add storage, functions, and more
+ */
+const backend = defineBackend({
+  auth,
+  storage,
+});
+
+// custom storage configurations
+const s3Bucket = backend.storage.resources.bucket;
+const cfnBucket = s3Bucket.node.defaultChild as s3.CfnBucket;
+
+const randomBucketName = `dots.in.name-${uuidv4()}`;
+
+cfnBucket.bucketName = randomBucketName;
+
+backend.storage.resources.cfnResources.cfnBucket.corsConfiguration = {
```

Comment:
*suggestion:* I would include a comment since it may not be obvious why this is done in the future.

```suggestion
// required to add the metadata header, which amplify-backend does not support
backend.storage.resources.cfnResources.cfnBucket.corsConfiguration = {
```